docs/design/wildcard-namespace-design.md

```diff
@@ -0,0 +1,98 @@
+# Design proposal: Wildcard Namespace Support for Velero Backup CLI
+
+## Abstract
+
+This proposal enables full wildcard namespace support for Velero backup CLI commands to resolve patterns like `prod-*` or `*-staging` into actual namespace names during command execution.
+Currently, wildcard patterns are stored literally in backup specs causing backup failures when the system tries to list resources in non-existent namespaces.
+
+## Background
+
+Velero's backup CLI currently accepts wildcard patterns for namespace inclusion and exclusion but has incomplete implementation.
+When users run commands like `velero backup create test --include-namespaces "prod-*"`, the wildcard pattern is stored literally in the backup specification without being resolved to actual namespace names.
+
+This causes backup failures because the system attempts to list resources in namespaces literally named `"prod-*"` which don't exist.
+The disconnect occurs between namespace collection (which uses proper glob matching) and resource collection (which uses literal patterns), resulting in inconsistent behavior and silent failures.
+
+## Goals
+
+- Enable wildcard patterns in backup CLI commands to work correctly by resolving them to actual namespace names
+- Provide clear error messages when wildcard patterns are invalid or match no namespaces
+- Maintain backward compatibility with existing exact namespace specifications
+
+## Non Goals
+
+- Wildcard support for restore operations (future enhancement)
+- Complex regex patterns beyond standard glob patterns
+- Dynamic namespace matching during backup execution (patterns resolved at CLI time only)
+
+## High-Level Design
+
+The solution resolves wildcard patterns during CLI command processing before creating backup objects.
+When a user specifies patterns like `prod-*`, the CLI will query the Kubernetes API to fetch all active namespaces, apply glob pattern matching, and store only the resolved actual namespace names in the backup specification.
```

Comment:
Why not keep the queried NS/matched NS in-memory instead of patching the backup spec ?

Reply:
When you say in-memory, do you mean that the spec would keep the wildcard namespaces, and that it would resolve at the time of backup creation, rather than us manually enumerating, finding and filling those namespaces in?
My approach was more find at cli command runtime and substitute. That makes it deterministic but then its not dynamic, like schedules picking up new namespaces matching the pattern

Reply:
This isolates the wildcard expansion to backup creation time (for CLI and hopefully for schedules too)